src/browser/web-card.js

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+'use strict';
+
+import {Record, Union, List, Maybe, Any} from 'typed-immutable';
```

Comment:
:+1: for ES Modules. Is this going to cause any problems with other modules that are AMD?